rlog/2023-08-04-wakurtosis-waku-scallability.mdx

```diff
@@ -0,0 +1,85 @@
+
+## Examining the Waku Protocol
+
+### Simulation Setup
+To evaluate Waku under varied conditions, we conducted simulations across a range of network sizes, topologies, and message rates. 
+Each simulation lasted 3 hours to rreach a steady state. 
+The network sizes explored included *75*, *150*, *300*, and *600* nodes. 
+For non-Discv5 simulations, we used static topologies with average node degrees of *K=3*, *K=13*, and *K=50*. 
+In simulations with Discv5, we set the *max_peers* parameter to *12* and *50* to approximate similar average degrees. 
+To stress test message throughput, we simulated message rates of *1*, *10*, and *100* messages per second. 
+This combination of network sizes, topologies, message rates, and hardware configurations enabled us to comprehensively evaluate Waku's performance and scalability boundaries under diverse conditions.
+
+### Results
+To provide a clear visualization of bandwidth usage in all the cases we studied, we've presented the results as multipliers over the pure payload for both reception (Rx) and transmission (Tx) in two heatmaps:
+
+![Bandwidth effiencies multipliers over pure payloads in transmission](/static/img/wakurtosis_waku/waku_bandwidth_rx_efficiency.png)
+![Bandwidth efficiencies multipliers over pure payloads in reception](/static/img/wakurtosis_waku/waku_bandwidth_tx_efficiency.png)
+
+Despite certain scalability challenges, Wakurtosis has proved effective in simulating a Waku network with over *600* nodes on a single machine. 
+Our analysis under baseline conditions --- static network with average degrees of *K=3*, *K=13*, and *K=50* --- revealed good scalability and stability from a memory standpoint. 
+However, there was evidence of increased memory usage with higher message rates, indicating potential areas for optimization. 
+Bandwidth utilization exhibited minor fluctuations as the network size increased, and efficiency in bandwidth usage improved with an increase in the message rate. 
+Some inefficiencies were noted in larger networks, suggesting room for improvement but overall Waku displayed very stable behaviour in terms of memory and bandwidth usage.
+
+#### Impact of Discovery v5
+To shed light on the effects of Discovery v5, we focus on the denser configurations for both non-Discv5 and Discv5 with *K=50*.
+The addition of Discv5 leads to higher memory consumption, especially as the network size increases.
+This is likely due to the additional routing information that needs to be stored and maintained.
+When examining the effects on bandwidth utilization, we see varying results.
+For reception rates, with lower message rates we observe that Discv5 consistently utilizes more bandwidth than the corresponding non-Discv5 case.
+However, those differences become less pronounced with higher message rates.
+For transmission rates, both cases show similar performance and both exhibit huge improvements in transmission efficiency at higher message loads.
```

Comment:
One more note, to reiterate: we wouldn't expect to see an efficiency that's higher than around 1/6. Where this is the case, we should assume there might be losses due to infrastructure (or other) limits.